conformance/lang/expressions/table_constructor_expr.balt

```diff
@@ -0,0 +1,1482 @@
+             Test for when all the members are of the same type.
+Labels: key-specifier, table-constructor-expr, table-type, record-type, string, mapping-constructor-expr, var
+
+function init() {
+    var v1 = table key(name) [
+        {name: "Jo"}
+    ];
+    table<record {|string name;|}> t1 = v1;
+    table<record {}> t = t1;
+
+    io:println(v1.length()); // @output 1
+    io:println(t is table<record {|readonly string name;|}> key(name)); // @output true
+
+    var v2 = table key(name) [
+        {name: "Jo"},
+        {name: "Amy"}
+    ];
+    table<record {|string name;|}> t2 = v2;
+    t = t2;
+
+    io:println(v2.length()); // @output 2
+    io:println(t is table<record {|string name;|}> key(name)); // @output true
+    io:println(t is table<record {|readonly string name;|}> key(name)); // @output true
+    io:println(t is table<record {|readonly string name;|}>); // @output true
+}
+
+Test-Case: output
+Description: Test when there is no contextually-expected type the member type of the inherent type being derived from the the static type of the expressions for the members, for a table constructor expression with a key-specifier. The member type will be the smallest record type that is a supertype of the static types of all the expressions in the row-list.
+             Test for when members are of different types.
+Labels: key-specifier, table-constructor-expr, table-type, record-type, int, string, boolean, mapping-constructor-expr, var
+
+function init() {
+    var v1 = table key (code) [
+        {code: 1, name: "Jo"},
+        {code: 2, id: 3},
+        {code: 3, employed: true}
+    ];
+    table<record {|readonly int code; string name?; int id?; boolean employed?;|}> key(code) _ = v1;
+    table<record {|int code; string name?; int id?; boolean employed?;|}> t1 = v1;
+    table<record {}> t = t1;
+
+    io:println(v1.length()); // @output 3
+    io:println(t is table<record {|string name?; int id?; boolean employed?; int code;|}>); // @output true
+    io:println(t is table<record {|string name?; int id?; boolean employed?; readonly int code;|}> key(code)); // @output true
+    io:println(t is table<record {|string name; int id; boolean employed;|}>); // @output false
+}
+
+Test-Case: output
+Description: Test when there is no contextually-expected type the member type of the inherent type being derived from the the static type of the expressions for the members, for a table constructor expression with a key-specifier. The member type will be the smallest record type that is a supertype of the static types of all the expressions in the row-list.
+             Test for when members are of different types and the mapping constructor has variable-name fields.
+Labels: key-specifier, table-constructor-expr, table-type, record-type, int, string, mapping-constructor-expr, var
```

Comment:
` boolean-literal`, `boolean`, `is-expr`

Reply:
I think we decided boolean covers boolean-literal.